rj_gameplay/rj_gameplay/tactic/pass_tactic.py

```diff
@@ -106,44 +106,98 @@ def __call__(
+        if self.passer_robot is not None and robot.id != self.passer_robot.id:
+            pass_dist = np.linalg.norm(passer_robot.pose[0:2] -
+                                       robot.pose[0:2])
+            goal_to_receiver = np.linalg.norm(robot.pose[0:2] -
+                                              rc.Field.their_goal_loc)
+            cost = 0
+            for enemy in world_state.their_robots:
+                cost -= 10 * np.linalg.norm(enemy.pose[0:2] - robot.pose[0:2])
+        else:
+            return 1e9
 
-        # TODO: should be dist in sec
-        # raw_dist = np.linalg.norm(robot.pose[0:2] - self.target_point)
-        # cost = cost + raw_dist
-        return cost
+    def unassigned_cost_fn(
+        self,
+        prev_result: Optional["RoleResult"],
+        world_state: rc.WorldState,
+    ) -> float:
 
-        if robot is None or self.target_point is None:
-            return 99
-        # TODO (#1669)
-        if not robot.visible:
-            return 99
-        if self.passer_robot is not None and robot.id == self.passer_robot.id:
-            # can't pass to yourself
-            return 99
+        #TODO: Implement real unassigned cost function
+        return role.BIG_STUPID_NUMBER_CONST_FOR_UNASSIGNED_COST_PLS_CHANGE
 
-        # TODO: pick "most open" pass
-        cost = 0 
-        for enemy in world_state.their_robots:
-            cost -= 10*np.linalg.norm(enemy.pose[0:2] - robot.pose[0:2])
 
-        # TODO: should be dist in sec
-        # raw_dist = np.linalg.norm(robot.pose[0:2] - self.target_point) 
-        # cost = cost + raw_dist
-        return cost
+class PassToBestReceiver(role.CostFn):
+    """
+    A cost function for how to choose a robot to pass to
+
+    """
+    def __init__(self, passer_robot: rc.Robot = None):
+        self.passer_robot = passer_robot
+        # self.chosen_receiver = None
+
+    def __call__(self, robot: rc.Robot, prev_result: Optional["RoleResult"],
+                 world_state: rc.WorldState) -> float:
+        if world_state is not None:
+            min_dist = 999
+            for our_robot in world_state.our_robots:
+                dist = np.linalg.norm(world_state.ball.pos -
+                                      our_robot.pose[0:2])
+                if min_dist > dist:
+                    min_dist = dist
+                    self.passer_robot = our_robot
+        if robot is None:
+            return 1e9
+        # if not robot.visible:
+        #     return 1e9
+        if self.passer_robot is not None and robot.id == self.passer_robot.id:
+            # can't pass to yourself
+            return 1e9
+        angle_threshold = 5
+        dist_threshold = 1.5
+        backpass_punish_weight = 0.5
+        if robot.id != self.passer_robot.id:
+            pass_dist = np.linalg.norm(self.passer_robot.pose[0:2] -
+                                       robot.pose[0:2])
+
+            cost = 0
+            goal_to_receiver = np.linalg.norm(robot.pose[0:2] -
```

Comment:
add a few comments explaining chunks of below code, e.g. here I would put 
"add to cost for backwards passes (scaled by backpass_punish_weight)"